From a1eb50653080e287ff981ecc88d0317f80d2048a Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 13 Aug 2018 15:55:09 +0800
Subject: [PATCH] driver: ethernet: phy: improve phy auto-negotiation
 configure

Configure fiber link only when phy supports fiber mode.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/phy/marvell.c |   16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/net/phy/marvell.c b/drivers/net/phy/marvell.c
index 55b8ca9..8d30f20 100644
--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -811,14 +811,16 @@ static int m88e1510_config_aneg(struct phy_device *phydev)
 	if (err < 0)
 		goto error;
 
-	/* Then the fiber link */
-	err = phy_write(phydev, MII_MARVELL_PHY_PAGE, MII_M1111_FIBER);
-	if (err < 0)
-		goto error;
+	if ((phydev->supported & SUPPORTED_FIBRE)) {
+		/* Then the fiber link */
+		err = phy_write(phydev, MII_MARVELL_PHY_PAGE, MII_M1111_FIBER);
+		if (err < 0)
+			goto error;
 
-	err = marvell_config_aneg_fiber(phydev);
-	if (err < 0)
-		goto error;
+		err = marvell_config_aneg_fiber(phydev);
+		if (err < 0)
+			goto error;
+	}
 
 	return phy_write(phydev, MII_MARVELL_PHY_PAGE, MII_M1111_COPPER);
 
-- 
1.7.9.5

